controllers/topology/current_state.go

```diff
@@ -18,12 +18,142 @@ package topology
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// Gets the current state of the Cluster topology.
-func (r *ClusterReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster) (*clusterTopologyState, error) {
-	// TODO: add get class logic; also remove nolint exception from clusterTopologyState and machineDeploymentTopologyState
-	return nil, nil
+// getCurrentState gets information about the current state of a Cluster by inspecting the state of the InfrastructureCluster,
+// the ControlPlane, and the MachineDeployments associated with the Cluster.
+func (r *ClusterReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster, class clusterv1.ClusterClass) (*clusterTopologyState, error) {
```

Comment:
```suggestion
func (r *ClusterReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster, class *clusterv1.ClusterClass) (*clusterTopologyState, error) {
```

Unless I missed something, seems a good change to avoid copying the class around?

Reply:
changed